compiler/control/OMROptions.cpp

```diff
@@ -2245,6 +2245,11 @@ OMR::Options::jitLatePostProcess(TR::OptionSet *optionSet, void * jitConfig)
             TR::Options::createDebug();
          }
 
+      if (!TR::Options::getCmdLineOptions()->isAnyReductionAlgorithmSet())
```

Comment:
Does this make sense for OMR? Does Max make sense as a language agnostic default? I'm trying to decide this - it is not clear to me what the answer is. I would say either there is a reasonable default or we should have a function that asserts as unimplemented so that if languages opt into this they are required to codify their default reduction algorithm.

If you could at least explain your choice and why it is a reasonable, language-agnostic default that would help.

Reply:
For various benchmarks with gencon, SUM produced the best results. For various benchmarks with balanced, MAX produced the best and most reliable results. MAX is also the middle ground between SUM and AVERAGE with regards to how conservative the compiler will be with reporting hot fields. For this reason, and since realistically dynamic breadth first scan ordering will be used primarily with balanced for the foreseeable future, (even though dynamic breadth scan ordering is directly competitive with hierarchical scan ordering in gencon), I had decided to make MAX the default. However I do believe that anyone using dynamic breadth first should try varying dbfMaxDepth and varying the 3 reduction algorithms to see what produces the best result for the application. With this information, can this remain as is or what would be your recommendation? @andrewcraik

Reply:
Ok if it is a reasonable default as you say then I guess I am fine with that.